### PR TITLE
Use Segoe UI for Latin characters before falling back to YaHei

### LINKED ISSUE #3237: Include Segoe UI in font list before YaHei
Microsoft YaHei is the default font used by gitea on Windows PCs, and contains the needed support for non-English characters such as those in the Chinese unicode range. Unfortunately, YaHei is not correctly hinted and results in bad aliasing on Windows machines when rendering English text via DirectWrite/Direct2D (tested with both Edge and Chrome under Windows 10).

The CSS standard actually has a very simple workaround for this, if a `font-family` stack is provided and any individual character is not found in the first-chosen font, the second in the list is tried _only for the characters that are not found_, and so on and so forth.

That means that simply changing the CSS declaration in `_base.less` from 

```css
    font-family: "Helvetica Neue", "Microsoft YaHei", Arial, Helvetica, sans-serif !important;
```
to 

```css
    font-family: "Helvetica Neue", "Segoe UI", "Microsoft YaHei", Arial, Helvetica, sans-serif !important;
```

will result in the browser automatically using Segoe UI (which is wonderfully hinted and has beautiful antialiasing under all browsers and platforms) for characters supported by that font, and Microsoft YaHei when characters outside of Segoe UI's supported range are used.

Here is a screenshot of the default/current Gitea configuration with Microsoft YaHei used for all text:

all-in-yahei

And here's an example with a mix of Segoe UI and Microsoft YaHei:

english-segoe-ui

In the second screenshot, the Chinese text remains rendered in Microsoft YaHei, but the English text benefits from the well-hinted Segoe UI glyphs.

An additional option (though much less straightforward) that many are not aware of is manually crafting a font face that defines which unicode characters are pulled from which font:

```css
@font-face {
    font-family: AsciiSegoe;
    src: local('Segoe UI'), local('SegoeUI')
    unicode-range: U+00-FF;
}

font-family: "Helvetica Neue", AsciiSegoe, "Microsoft YaHei", Arial, Helvetica, sans-serif !important;
```

The above would load only basic ASCII characters from Segoe UI, and move on to the next entry in the font-family list for all other characters. Obviously that can be flipped around to load only Chinese characters from YaHei and everything else from the default system font, or however else you wish to configure it.

### PR DESCRIPTION
**Pull request by mqudsi**
This fixes issues with character rendering on hidpi displays and other
aliasing/hinting issues, as Segoe UI has been painstakingly hinted by
typographers for Microsoft while YaHei has not. See #3237 for more
details.

Closes #3237

### CODE REVIEW

**Comment by mqudsi** [comment]
Note that I didn't know whether to include the `lessc`-transformed css output in the commit as build results are normally not in the repo, but the transformed css files are.

**Comment by thehowl** [comment]
The resulting index.css should be committed, actually. You can just `make generate-stylesheets` to call that.

File: public/less/_base.less [hunk 1]
```diff
@@ -1,7 +1,7 @@
 @footer-margin: 40px;
 
 body {
-    font-family: "Lato", "Microsoft YaHei", Arial, Helvetica, sans-serif !important;
+    font-family: "Lato", "Segoe UI", "Microsoft YaHei", Arial, Helvetica, sans-serif !important;
```
Comment (thehowl) [inline public/less/_base.less:4]: What are the latin characters affected by this? Does Lato not have pretty much every latin character you can think of?

Comment (mqudsi) [reply public/less/_base.less:4]: #3237 was actually filed before the switch to `Lato`, but if `Lato` isn't installed the issue remains.

Comment (thehowl) [reply public/less/_base.less:4]: I guess the lato fontface was never added. Fair enough, in any case.

**Review by thehowl — APPROVED** [review]

**Comment by mqudsi** [comment]
@lunny I assumed it had to do with my earlier question about the LESS/CSS and per the logs (and that does seem to be the case). However I just committed the transformed CSS [but the CI has failed again](https://drone.gitea.io/go-gitea/gitea/3931/5). I can't tell if the `sh: write error: Resource temporarily unavailable` is normal?

I ran the same test locally (`gmake stylesheets-check`) and it passes?

**Comment by thehowl** [comment]
Yeah, drone probably built the wrong commit. @lunny can you restart build?

**Comment by lunny** [comment]
@mqudsi @thehowl restarted and build Successfully.

**Review by appleboy — APPROVED** [review]

**Comment by mqudsi** [comment]
Thanks @lunny @thehowl

**Pull request merged**